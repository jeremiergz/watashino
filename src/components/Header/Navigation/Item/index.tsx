import React, { ComponentType } from 'react';
import Box from '../../../primitives/Box';
import Flex from '../../../primitives/Flex';
import Link from '../../../widgets/Link';

type ItemProps = {
    link: {
        icon: ComponentType;
        name: string;
        to: string;
    };
};

const Item = ({ link, ...rest }: ItemProps) => {
    return (
        <Flex alignItems="center" flexDirection="column" justifyContent="center" minWidth={96} {...rest}>
            <Link alignItems="center" display="flex" flexDirection="column" to={link.to}>
                <Box as={link.icon} height={32} width={32} />
                {link.name}
            </Link>
        </Flex>
    );
};

Item.displayName = 'Item';

export { ItemProps };
export default Item;
