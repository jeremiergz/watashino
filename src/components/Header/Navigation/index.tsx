import Link from 'components/Link';
import FlexBox from 'components/primitives/FlexBox';
import { useTheme } from 'components/providers/ThemeProvider';
import SVG from 'components/SVG';
import BookIcon from 'components/svgs/icons/Book';
import DescriptionIcon from 'components/svgs/icons/Description';
import HouseIcon from 'components/svgs/icons/House';
import MoodIcon from 'components/svgs/icons/Mood';
import { graphql, useStaticQuery } from 'gatsby';
import { trackCustomEvent } from 'gatsby-plugin-google-analytics';
import React from 'react';
import { Routes } from 'utils/Routes';

const iconsMapping = {
  book: BookIcon,
  home: HouseIcon,
  mood: MoodIcon,
};

const Navigation: React.FC = () => {
  const { theme } = useTheme();
  const {
    allMarkdownRemark: { totalCount },
    allNavigationJson: { nodes: links },
    dataJson: { resumeLink },
  } = useStaticQuery<GraphQL.NavigationDataQuery>(graphql`
    query NavigationData {
      allMarkdownRemark {
        totalCount
      }
      allNavigationJson {
        nodes {
          icon
          ignoreInNavigation
          keywords
          name
          navOrder
          to
        }
      }
      dataJson {
        resumeLink
      }
    }
  `);
  const handleResumeClick = () => {
    trackCustomEvent({
      action: 'click',
      category: 'Open Resume',
    });
  };
  return (
    <FlexBox as="nav" flexWrap="wrap" justifyContent="center" marginTop={{ _: 16, tablet: 0 }}>
      {links
        .filter(link => {
          if (link.to === Routes.posts && totalCount === 0) return false;
          return !link.ignoreInNavigation;
        })
        .sort((a, b) => a.navOrder - b.navOrder)
        .map(link => (
          <FlexBox alignItems="center" flexDirection="column" justifyContent="center" key={link.name} minWidth={96}>
            <Link alignItems="center" display="flex" flexDirection="column" partiallyActive to={link.to}>
              <SVG as={iconsMapping[link.icon]} fill={theme.colors.text} />
              {link.name}
            </Link>
          </FlexBox>
        ))}
      <FlexBox alignItems="center" flexDirection="column" justifyContent="center" minWidth={96}>
        <Link
          alignItems="center"
          display="flex"
          external
          flexDirection="column"
          onClick={handleResumeClick}
          to={resumeLink}
        >
          <SVG as={DescriptionIcon} fill={theme.colors.text} />
          Resume
        </Link>
      </FlexBox>
    </FlexBox>
  );
};

Navigation.displayName = 'Navigation';

export default Navigation;
