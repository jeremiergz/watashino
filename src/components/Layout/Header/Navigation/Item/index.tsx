import React, { ComponentType } from 'react';
import styled from 'styled-components';
import BaseLink from '../../../../widgets/Link';

type Props = {
    link: {
        icon: ComponentType;
        name: string;
        to: string;
    };
};

const Item = styled.div`
    min-width: 96px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
`;

const Link = styled(BaseLink)`
    text-align: center;
`;

const Svg = styled.svg`
    height: 32px;
    width: 32px;
`;

const ItemComponent = ({ link, ...rest }: Props) => {
    return (
        <Item {...rest}>
            <Link to={link.to}>
                <Svg as={link.icon} />
                <div>{link.name}</div>
            </Link>
        </Item>
    );
};

ItemComponent.displayName = 'Item';

export default ItemComponent;
