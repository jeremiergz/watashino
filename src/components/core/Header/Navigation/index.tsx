import { graphql, useStaticQuery } from 'gatsby';
import React, { ReactNode } from 'react';
import Anchor from '../../../common/Anchor';
import Link from '../../../common/Link';
import Box from '../../../primitives/Box';
import Flex from '../../../primitives/Flex';
import BookIcon from '../../../svg/Book';
import DownloadIcon from '../../../svg/Download';
import HouseIcon from '../../../svg/House';
import MoodIcon from '../../../svg/Mood';
import styled from 'styled-components';

const iconsMapping = {
  book: BookIcon,
  home: HouseIcon,
  mood: MoodIcon,
};

const LinkContainer = ({ children, ...rest }: { children: ReactNode }) => (
  <Flex alignItems="center" flexDirection="column" justifyContent="center" minWidth={96} {...rest}>
    {children}
  </Flex>
);

const Svg = styled(Box)`
  fill: ${({ theme }) => theme.svg};
`;

const Navigation = () => {
  const {
    allMarkdownRemark: { totalCount },
    allNavigationJson: { nodes: links },
    dataJson: { resumeFile },
  } = useStaticQuery<GraphQL.NavigationDataQuery>(graphql`
    query NavigationData {
      allMarkdownRemark {
        totalCount
      }
      allNavigationJson {
        nodes {
          icon
          ignoreInNavigation
          keywords
          name
          navOrder
          to
        }
      }
      dataJson {
        resumeFile
      }
    }
  `);
  return (
    <Flex as="nav" marginTop={{ _: 16, tablet: 0 }}>
      {links
        .filter(link => {
          if (link.to === '/posts' && totalCount === 0) return false;
          return !link.ignoreInNavigation;
        })
        .sort((a, b) => a.navOrder - b.navOrder)
        .map(link => (
          <LinkContainer key={link.name}>
            <Link alignItems="center" display="flex" flexDirection="column" partiallyActive to={link.to}>
              <Svg as={iconsMapping[link.icon]} height={32} width={32} />
              {link.name}
            </Link>
          </LinkContainer>
        ))}
      <LinkContainer>
        <Anchor alignItems="center" display="flex" download flexDirection="column" href={resumeFile}>
          <Svg as={DownloadIcon} height={32} width={32} />
          Resume
        </Anchor>
      </LinkContainer>
    </Flex>
  );
};

Navigation.displayName = 'Navigation';

export default Navigation;
