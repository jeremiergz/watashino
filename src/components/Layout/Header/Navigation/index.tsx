import React from 'react';
import styled from 'styled-components';
import theme from '../../../../theme';
import BookIcon from '../../../svg/Book';
import HouseIcon from '../../../svg/House';
import MoodIcon from '../../../svg/Mood';
import Item from './Item';

const Navigation = styled.div`
    display: flex;
    margin-top: 16px;
    ${theme.breakpoints.minWidth.tablet} {
        margin-top: 0;
    }
`;

const links = {
    home: {
        icon: HouseIcon,
        name: 'Home',
        to: '/',
        keywords: ['contact', 'gatsby', 'react'],
    },
    articles: {
        icon: BookIcon,
        name: 'Articles',
        to: '/articles',
        keywords: ['articles', 'gatsby', 'react'],
    },
    aboutMe: {
        icon: MoodIcon,
        name: 'About Me',
        to: '/about-me',
        keywords: ['about', 'gatsby', 'react'],
    },
    notFound: {
        name: '404 Not Found',
        to: '/',
        keywords: ['404', 'gatsby', 'react'],
        ignoreInNavigation: true,
    },
};

const NavigationComponent = () => {
    return (
        <Navigation>
            {Object.keys(links)
                .filter(key => !links[key].ignoreInNavigation)
                .map(key => {
                    const link = links[key];
                    return <Item key={link.name} link={link} />;
                })}
        </Navigation>
    );
};

NavigationComponent.displayName = 'Navigation';
NavigationComponent.links = links;

export default NavigationComponent;
