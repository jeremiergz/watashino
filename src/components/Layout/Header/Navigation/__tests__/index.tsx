import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Navigation from '..';

describe('components/Layout/Header/Navigation Test Suite', () => {
    it('matches snapshot', () => {
        const jsx = <Navigation />;
        expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
});
