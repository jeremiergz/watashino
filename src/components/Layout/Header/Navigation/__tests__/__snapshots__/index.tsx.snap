// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/Header/Navigation Test Suite matches snapshot 1`] = `
<div
  className="sc-ifAKCX fSDoTH"
>
  <div
    className="sc-bwzfXH wIIFn"
  >
    <a
      className="sc-htpNat hhQZsf sc-bdVaJa fkCkAu"
      href="/"
      underlined={false}
    >
      <svg
        className="sc-bxivhb btqxNa"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0H24V24H0z"
          fill="none"
        />
        <path
          d="M19 9.3V4h-3v2.6L12 3 2 12h3v8h5v-6h4v6h5v-8h3l-3-2.7zm-9 .7c0-1.1.9-2 2-2s2 .9 2 2h-4z"
        />
      </svg>
      Home
    </a>
  </div>
  <div
    className="sc-bwzfXH wIIFn"
  >
    <a
      className="sc-htpNat hhQZsf sc-bdVaJa fkCkAu"
      href="/articles"
      underlined={false}
    >
      <svg
        className="sc-bxivhb btqxNa"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0H24V24H0z"
          fill="none"
        />
        <path
          d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
        />
        <path
          d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
        />
      </svg>
      Articles
    </a>
  </div>
  <div
    className="sc-bwzfXH wIIFn"
  >
    <a
      className="sc-htpNat hhQZsf sc-bdVaJa fkCkAu"
      href="/about-me"
      underlined={false}
    >
      <svg
        className="sc-bxivhb btqxNa"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
        />
      </svg>
      About Me
    </a>
  </div>
</div>
`;
